crates/sui-types/src/deny_list_v2.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base_types::{EpochId, SuiAddress};
+use crate::deny_list_v1::{get_deny_list_root_object, DENY_LIST_COIN_TYPE_INDEX, DENY_LIST_MODULE};
+use crate::dynamic_field::{get_dynamic_field_from_store, DOFWrapper};
+use crate::id::UID;
+use crate::storage::ObjectStore;
+use crate::{MoveTypeTagTrait, SUI_FRAMEWORK_PACKAGE_ID};
+use move_core_types::ident_str;
+use move_core_types::language_storage::{StructTag, TypeTag};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+/// Rust representation of the Move type 0x2::config::Config.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Config {
+    id: UID,
+}
+
+/// Rust representation of the Move type 0x2::config::Setting.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct Setting<V: Copy + fmt::Debug> {
+    data: Option<SettingData<V>>,
+}
+
+/// Rust representation of the Move type 0x2::config::SettingData.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct SettingData<V: Copy + fmt::Debug> {
+    newer_value_epoch: u64,
+    newer_value: V,
+    older_value_opt: Option<V>,
+}
```

Comment:
Some of this is duplicated on my branch, which is fine. I will clean it up when I land mine